frontend/src/features/mapSideBar/SidebarContents/subforms/ClassificationForm.tsx

```diff
@@ -86,17 +90,37 @@ export const ClassificationForm: React.FC<IClassificationFormProps> = ({
   encumbranceField,
 }) => {
   const formikProps = useFormikContext();
+  const keycloak = useKeycloakWrapper();
   let surplusActiveOrEncumbered =
     getIn(formikProps.values, field) === Classifications.SurplusEncumbered ||
     getIn(formikProps.values, field) === Classifications.SurplusActive;
 
   /** classId based on current formik values to determine which classsification information box to display */
   let classId = getIn(formikProps.values, field);
 
+  /** determine how to filter the classifications */
+  const determineFilter = (val: Number) => {
+    /** non SRES users cannot see subdivided */
+    if (!keycloak.hasClaim(Claims.ADMIN_PROPERTIES)) {
+      return (
+        val !== Classifications.Subdivided &&
+        val !== Classifications.Disposed &&
+        val !== Classifications.Demolished
+      );
+    } else {
+      /** only buildings can be demolished */
+      if (!fieldLabel?.includes('Building')) {
+        return val !== Classifications.Disposed && val !== Classifications.Demolished;
+      } else {
+        return val !== Classifications.Disposed;
+      }
+    }
+  };
+
   let filteredClassifications = classifications;
   /** users not allowed to select disposed, but display these values if one of these classifications has already been selected. */
   filteredClassifications = classifications.filter(
-    c => Number(c.value) !== Classifications.Disposed || +c.value === +classId,
+    c => determineFilter(+c.value) || +c.value === +classId,
```

Comment:
Not sure if this is an issue or not, but be careful with unary + as it will convert some values to numbers unexpectedly.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus

For example if classId is '' (formik default) +classId will be converted to 0, which is equivalent to Core Operational.